lib/connections/service.go

```diff
@@ -1050,3 +1075,32 @@ func (s *service) validateIdentity(c internalConn, expectedID protocol.DeviceID)
 
 	return nil
 }
+
+type nextDialRegistry map[protocol.DeviceID]map[string]time.Time
+
+func (r nextDialRegistry) get(device protocol.DeviceID, addr string) (time.Time, bool) {
+	if _, ok := r[device]; !ok {
+		return time.Time{}, false
+	}
+	if _, ok := r[device][addr]; !ok {
+		return time.Time{}, false
+	}
+	return r[device][addr], true
```

Comment:
Also not really required as this could be shortened to

```
t := r[device][addr]
return t, !t.IsZero()
```

or just letting the caller handle the fact that the time might be zero -- which might happen by default this this effectively asks for a dial in the distant past?